src/caselawclient/models/documents.py

```diff
@@ -495,6 +497,14 @@ def enrich(self) -> None:
             enrich=True,
         )
 
+    def enrich(self, cooldown: datetime.timedelta = MINIMUM_REPARSE_TIME) -> bool:
```

Comment:
Is this method always triggered by a user interaction? I'm wondering if we might be worth having a `can_enrich?` method, which will ghost out / hide the enrich button if it returns false